jsonrpc/debug_endpoint.go

```diff
@@ -0,0 +1,359 @@
+package jsonrpc
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/0xPolygon/polygon-edge/helper/hex"
+	"github.com/0xPolygon/polygon-edge/state"
+	"github.com/0xPolygon/polygon-edge/state/runtime"
+	"github.com/0xPolygon/polygon-edge/state/runtime/tracer"
+	"github.com/0xPolygon/polygon-edge/types"
+)
+
+type debugBlockchainStore interface {
+	// Header returns the current header of the chain (genesis if empty)
+	Header() *types.Header
+
+	// GetHeaderByNumber gets a header using the provided number
+	GetHeaderByNumber(uint64) (*types.Header, bool)
+
+	// ReadTxLookup returns a block hash in which a given txn was mined
+	ReadTxLookup(txnHash types.Hash) (types.Hash, bool)
+
+	// GetBlockByHash gets a block using the provided hash
+	GetBlockByHash(hash types.Hash, full bool) (*types.Block, bool)
+
+	// GetBlockByNumber gets a block using the provided height
+	GetBlockByNumber(num uint64, full bool) (*types.Block, bool)
+
+	TraceMinedBlock(*types.Block, runtime.Tracer) ([]interface{}, error)
+
+	TraceMinedTxn(*types.Block, types.Hash, runtime.Tracer) (interface{}, error)
+
+	TraceCall(*types.Transaction, *types.Header, runtime.Tracer) (interface{}, error)
+}
+
+type debugTxPoolStore interface {
+	GetNonce(types.Address) uint64
+}
+
+type debugStateStore interface {
+	GetAccount(root types.Hash, addr types.Address) (*state.Account, error)
+}
+
+type debugStore interface {
+	debugBlockchainStore
+	debugTxPoolStore
+	debugStateStore
+}
+
+// Debug is the debug jsonrpc endpoint
+type Debug struct {
+	store debugStore
+}
+
+type TraceConfig struct {
```

Comment:
Add a test to make sure the encoding/decoding is as expected.

Reply:
Added c19bb53a5cf11c2783cc804bdaa4686869c6dfba. Other tests will be in other commits